.pre-commit-config.yaml

```diff
@@ -0,0 +1,18 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+      - id: trailing-whitespace
+        files: '\.py$'
+      - id: end-of-file-fixer
+        files: '\.py$'
+      - id: check-yaml
+      - id: check-added-large-files
+        files: '\.py$'
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
```

Comment:
I think black needs to be added back in -- ruff can auto fix errors but it doesn't do any formatting.

Reply:
Oh, just found out that it does format as well as doing the error checking. Awesome!